cowbird/services/impl/geoserver.py

```diff
@@ -18,30 +26,31 @@ class Geoserver(Service):
     """
     Keep Geoserver internal representation in sync with the platform.
     """
-    required_params = [SERVICE_URL_PARAM, SERVICE_WORKSPACE_DIR_PARAM]
+    required_params = [SERVICE_URL_PARAM, SERVICE_WORKSPACE_DIR_PARAM, SERVICE_ADMIN_USER, SERVICE_ADMIN_PASSWORD]
 
     def __init__(self, settings, name, **kwargs):
         # type: (SettingsType, str, dict) -> None
         """
-        Create the geoserver instance.
+        Create the geoserver service instance.
 
         @param settings: Cowbird settings for convenience
         @param name: Service name
         """
         super(Geoserver, self).__init__(settings, name, **kwargs)
+        self.api_url = "{}/rest".format(self.url)
+        self.auth = (self.admin_user, self.admin_password)
```

Comment:
These could be `None` based on initialization values. 
I'm not sure if requests using `auth` will like `(None, None)`, `("user", None)`, etc. 
Might need to set `None` directly if any is missing.

Reply:
I tested both, as well as (None, "password") and empty strings. Cowbird does not seem affected and the individual requests to Geoserver return the anticipated response, 401 : failed because lack of valid creds.

The structure did change since this comment, so it might not even be an issue anymore